canvas/canvas_usage.py

```diff
@@ -0,0 +1,109 @@
+import logging
+from json.decoder import JSONDecodeError
+from requests_futures.sessions import FuturesSession
+from concurrent.futures import as_completed
+import json
+logger = logging.getLogger(__name__)
+import pandas as pd
+
+
+class CanvasUsage:
+    def __init__(self, canvas_url, canvas_token, retry_attempts, canvas_ids):
+        self.canvas_url = canvas_url,
+        self.canvas_token = canvas_token,
+        self.canvas_ids = canvas_ids
+        self.retry_attempts = retry_attempts
+        self.canvas_usage_courses = []
+        self.course_retry_list = []
+        self.retry_count = 0
+
+    def parsing_canvas_usage_data(self, response):
+        logger.info("parsing_canvas_usage_data Call")
+        new_dic = {}
+        if response is None:
+            logger.info(f"For Canvas usage response is None ")
+            return
+
+        logger.info(f"CanvasUsage date collected so far : {len(self.canvas_usage_courses)}")
+        status = response.result().status_code
+
+        course_id = response.result().url.split('courses/')[1].split('/')[0]
+
+        status_codes_to_retry = [500]
```

Comment:
If you want to test how my retry is working add 404 to this list. it will try for 3 times and moves on